Add action to build dependencies table

aditigopalan opened the pull request:
Adding a github action to do build HTAN.dependencies.csv from the schematic API endpoints. Fixes #442

adamjtaylor commented on .github/workflows/build-dependencies-table.yml:
```diff
@@ -0,0 +1,61 @@
+
+on:
+  release:
+    types: [published]
```
Would there be advantages to doing this as part of our PR process for other data model changes? 

We could do it after JSON-LD generation. It might be a useful sniff test. That would require moving this job to be part of the `ci-lint-validate-convert` action and making it dependent on the JSON-LD creation being successful.

adamjtaylor commented on .github/workflows/build-dependencies-table.yml:
```diff
@@ -0,0 +1,61 @@
+        import pandas as pd
+
+        url = "https://schematic.api.sagebionetworks.org/v1/ui/"  # Endpoint URL
+        jsonld_url = "https://raw.githubusercontent.com/ncihtan/data-models/main/HTAN.model.jsonld"
```
If we were to move to being part of the `ci-lint-validate-convert` action we would need to set this URL dynamicaly

adamjtaylor reviewed (changes requested):
Looks good but have some questions about if we want this as a separate action or might be better run as part of the `lint-validate-convert` action

aditigopalan commented on .github/workflows/ci-lint-validate-convert.yml:
```diff
@@ -87,11 +82,79 @@ jobs:
+          import requests
+          import pandas as pd
+
+          url = "https://schematic.api.sagebionetworks.org/v1/ui/"  # Endpoint URL
+          if "${{ github.event_name }}" == "pull_request":
+              jsonld_url = "https://raw.githubusercontent.com/${{ github.repository }}/pull/${{ github.event.pull_request.number }}/head/HTAN.model.jsonld"
+          else:
+              jsonld_url = "https://raw.githubusercontent.com/${{ github.repository }}/main/HTAN.model.jsonld"
```
@adamjtaylor is this what you meant by creating dynamic links?

aditigopalan commented:
> We could do it after JSON-LD generation. It might be a useful sniff test. That would require moving this job to be part of the `ci-lint-validate-convert` action and making it dependent on the JSON-LD creation being successful.

I just added this workflow to `ci-lint-validate-convert`

Changes:
1. Added the build-dependencies job, which depends on the convert job.
2. Integrated the original steps for building the dependencies table into this new job.
3. Added a "Fetch latest changes" step in the build-dependencies job to ensure it pulls the latest changes from the main branch, including the newly created JSON-LD file.

Let me know if this works!
